crates/polars-time/src/truncate.rs

```diff
@@ -82,3 +82,59 @@ impl PolarsTruncate for DateChunked {
         Ok(out?.into_date())
     }
 }
+
+#[cfg(feature = "dtype-duration")]
+impl PolarsTruncate for DurationChunked {
+    fn truncate(&self, _tz: Option<&Tz>, every: &Utf8Chunked, offset: &str) -> PolarsResult<Self> {
+        let to_time_unit = match self.time_unit() {
+            TimeUnit::Nanoseconds => Duration::duration_ns,
+            TimeUnit::Microseconds => Duration::duration_us,
+            TimeUnit::Milliseconds => Duration::duration_ms,
+        };
+
+        let offset_duration = Duration::parse(offset);
+        if !offset_duration.is_constant_duration() {
+            polars_bail!(InvalidOperation:
+                "Cannot offset a Duration series by a non-constant duration."
+            );
+        }
+        let offset_units = to_time_unit(&offset_duration);
+
+        let out = if every.len() == 1 {
+            if let Some(every) = every.get(0) {
+                let every_duration = Duration::parse(every);
+                if every_duration.is_constant_duration() {
+                    let every_units = to_time_unit(&every_duration);
+
+                    Ok(self
+                        .0
+                        .apply_values(|duration| duration - duration % every_units + offset_units))
```

Comment:
same here - in Rust the result of modulo isn't always positive, e.g. `-1 % 7` is `-1` (not `6` as I would have expected) - I suspect something unpredictable will happen here with negative numbers, hence my request for a test

thanks 🙏